app/modules/translator.py

```
@@ -102,34 +142,34 @@ def translate_topics(self, topics: List[Dict[str, Any]]) -> List[Dict[str, Any]]
-            "topics": translated_topics
-        }
-
     def save_output(self, data: Dict[str, Any], output_file: str):
         with open(output_file, "w", encoding="utf-8") as f:
             json.dump(data, f, ensure_ascii=False, indent=2)
         print(f"\n✓ Translation complete! Output saved to: {output_file}")
 
-    def process(self, input_file: str, output_file: str):
-        print(f"Reading data from: {input_file}")
-        topics = self.read_input_data(input_file)
+    def process(self, output_file: str, input_file: str = None, input_data: List[Dict[str, Any]] = None):
```

Ruff (0.14.8)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)